testgen/src/light_block.rs

```diff
@@ -0,0 +1,252 @@
+use gumdrop::Options;
+use serde::{Deserialize, Serialize};
+use simple_error::*;
+
+use crate::helpers::parse_as;
+use crate::validator::generate_validators;
+use crate::{Commit, Generator, Header, Validator};
+use std::str::FromStr;
+use tendermint::block::signed_header::SignedHeader;
+use tendermint::node::Id as PeerId;
+use tendermint::validator;
+use tendermint::validator::Set as ValidatorSet;
+
+/// A light block is the core data structure used by the light client.
+/// It records everything the light client needs to know about a block.
+/// NOTE: This struct & associated `impl` below are a copy of light-client's `LightBlock`.
+/// The copy is necessary here to avoid a circular dependency.
+/// Cf. https://github.com/informalsystems/tendermint-rs/issues/605
+/// TODO: fix redundant code without introducing cyclic dependency.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct TMLightBlock {
+    /// Header and commit of this block
+    pub signed_header: SignedHeader,
+    /// Validator set at the block height
+    pub validators: ValidatorSet,
+    /// Validator set at the next block height
+    pub next_validators: ValidatorSet,
+    /// The peer ID of the node that provided this block
+    pub provider: PeerId,
+}
+
+impl TMLightBlock {
+    /// Constructs a new light block
+    pub fn new(
```

Comment:
This function is not used anywhere. Should we keep it?

Reply:
I removed it for now :)